### PR TITLE
Fix ext authz deny all traffic if encountering error with the provider

### PR DESCRIPTION
**Pull request by hanxiaop**
**Please provide a description of this PR:**
Fix https://github.com/istio/istio/issues/46951.

I think this will loose the rbac, however it makes sense to only deny rules that are delegated to the ext authz provider.

### CODE REVIEW

File: pilot/pkg/security/authz/builder/builder.go [hunk 1]
```diff
@@ -352,6 +354,19 @@ func (b Builder) buildTCP(rules *rbacpb.RBAC, shadowRules *rbacpb.RBAC, provider
...
 
+func getBadCustomDenyRules(rules *rbacpb.RBAC) *rbacpb.RBAC {
+	rbac := &rbacpb.RBAC{
+		Action:   rbacpb.RBAC_DENY,
+		Policies: map[string]*rbacpb.Policy{},
+	}
+	keys := maps.Keys(rules.Policies)
+	sort.Strings(keys)
+	for i, key := range keys {
+		rbac.Policies[fmt.Sprintf(badCustomActionTemplate, i)] = rules.Policies[key]
```
Comment (hzxuzhonghu) [inline pilot/pkg/security/authz/builder/builder.go:365]: Each rule has a name. Can we append the name with `-deny-due-to-bad-CUSTOM-action`

Comment (hzxuzhonghu) [reply pilot/pkg/security/authz/builder/builder.go:365]: Maybe simply set rules.Action to deny without changing the policy name

Comment (hanxiaop) [reply pilot/pkg/security/authz/builder/builder.go:365]: I think adding a name like `deny-due-to-bad-CUSTOM-action` may be better for users to understand the denied rules, so I prefer the first option.

Comment (hzxuzhonghu) [reply pilot/pkg/security/authz/builder/builder.go:365]: Sounds good

Comment (hanxiaop) [reply pilot/pkg/security/authz/builder/builder.go]: The name works. I haven't found any length limit of the policy name, however I'm not sure if the long name like this matters or not:
```json
{
  "name": "envoy.filters.http.rbac",
  "typed_config": {
    "@type": "type.googleapis.com/envoy.extensions.filters.http.rbac.v3.RBAC",
    "rules": {
      "action": "DENY",
      "policies": {
        "istio-ext-authz-ns[istio-system]-policy[ext-authz]-rule[0]-deny-due-to-bad-CUSTOM-action": {
          "permissions": [
            {
              "and_rules": {
                "rules": [
                  {
                    "or_rules": {
                      "rules": [
                        {
                          "url_path": {
                            "path": {
                              "exact": "/headers"
                            }
                          }
                        }
                      ]
                    }
                  }
                ]
              }
            }
          ],
          "principals": [
            {
              "and_ids": {
                "ids": [
                  {
                    "any": true
                  }
                ]
              }
            }
          ]
        }
      }
    }
  }
}

```

Comment (hzxuzhonghu) [reply pilot/pkg/security/authz/builder/builder.go]: The naming is already long, not matter.

With this we can know which authz policy not work

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**